### PR TITLE
added toArrayOf and toArrayOfOn

### PR DESCRIPTION
**Pull request by dwhitney**
## What does this pull request do?

I've added two functions `Data.Lens.Fold (toArrayOf, toArrayOfOn)` since `Array` is a fairly common data structure compared to `List`, which is already supported

## Where should the reviewer start?

The functions are in `Fold.purs` and then re-exported in `Data.Lens`

## How should this be manually tested?

I added an example in `test/Main.purs` that combines a couple of Prisms with `traversed`

### CODE REVIEW

File: src/Data/Lens/Fold.purs [hunk 1]
```diff
@@ -148,6 +149,15 @@ toListOfOn s p = toListOf p s
 
 infixl 8 toListOfOn as ^..
 
+-- | Collects the foci of a `Fold` into an array.
+toArrayOf :: forall s t a b. Fold (Endo (->) (Array a)) s t a b -> s -> Array a
+toArrayOf p = foldrOf p Array.cons []
```
Comment (hdgarrood) [inline src/Data/Lens/Fold.purs]: This is going to be quadratic in the length of the array being constructed. I think `Array.fromFoldable <<< toListOf p` would be a better implementation. (There may well be a better one still, though.)

Comment (dwhitney) [reply src/Data/Lens/Fold.purs]: ok I can make that change

Comment (dwhitney) [reply src/Data/Lens/Fold.purs]: alright it's updated

**Review by hdgarrood — APPROVED** [review]
Seems sensible to me. There may be a better still way of implementing this, but this implementation is fine for now I think, and these functions are sensible things to include imo.

**Comment by thomashoneyman** [comment]
Thanks, @dwhitney!

**Pull request merged**